src/platform_impl/macos/util/mod.rs

```diff
@@ -63,3 +65,8 @@ pub fn window_position(position: LogicalPosition<f64>) -> NSPoint {
         CGDisplay::main().pixels_high() as CGFloat - position.y as CGFloat,
     )
 }
+
+pub fn cursor_position() -> crate::dpi::PhysicalPosition<f64> {
+    let pt: NSPoint = NSEvent::mouseLocation();
```

Comment:
Huh, I would have expected this to be main-thread only, but I think it's actually fine

Reply:
I actually don't have a macOS to test this, so I have to take your word for it.